File: packages/typescript/src/code-toolbox/WorkspaceTsCodeToolbox.ts
import { WorkspaceCodeToolbox } from '../Workspace'

export class WorkspaceTsCodeToolbox implements WorkspaceCodeToolbox {
  public getDefaultImage(): string {
    return 'daytonaio/sdk-typescript:v0.49.0-3'
  }

  public getRunCommand(code: string): string {
    const base64Code = Buffer.from(code).toString('base64')
    return `sh -c 'echo ${base64Code} | base64 --decode | npx ts-node --compiler-options "{\\\"module\\\":\\\"CommonJS\\\"}" 2>&1 | grep -vE "npm notice|npm warn exec"'`
  }
}

File: packages/python/src/daytona_sdk/code_toolbox/workspace_ts_code_toolbox.py
import base64
from typing import Optional
from ..common.code_run_params import CodeRunParams


class WorkspaceTsCodeToolbox:
    def get_run_command(self, code: str, params: Optional[CodeRunParams] = None) -> str:
        # Encode the provided code in base64
        base64_code = base64.b64encode(code.encode()).decode()

        # Build environment variables string
        env_vars = ""
        if params and params.env:
            env_vars = ' '.join(f"{key}='{value}'" for key, value in params.env.items())

        # Build command-line arguments string
        argv = ""
        if params and params.argv:
            argv = ' '.join(params.argv)

        # Combine everything into the final command for TypeScript
        return f""" sh -c 'echo {base64_code} | base64 --decode | {env_vars} npx ts-node -O "{{\\\"module\\\":\\\"CommonJS\\\"}}" -e "$(cat)" x {argv} 2>&1 | grep -vE "npm notice|npm warn exec"' """
